packages/dev/loaders/src/glTF/2.0/Extensions/Interactivity/interactivityFunctions.ts

```diff
@@ -0,0 +1,230 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { IKHRInteractivity, IKHRInteractivity_Configuration, IKHRInteractivity_Node, IKHRInteractivity_Variable } from "babylonjs-gltf2interface";
+import type { IFlowGraphBlockConfiguration } from "core/FlowGraph";
+import type { ISerializedFlowGraph, ISerializedFlowGraphBlock, ISerializedFlowGraphConnection, ISerializedFlowGraphContext } from "core/FlowGraph/typeDefinitions";
+import { RandomGUID } from "core/Misc";
+import { gltfPropertyNameToBabylonPropertyName, gltfToFlowGraphTypeMap, gltfTypeToBabylonType } from "./utils";
+
+function convertValueWithType(configObject: IKHRInteractivity_Configuration, definition: IKHRInteractivity) {
+    if (configObject.type !== undefined) {
+        // get the type on the gltf definition
+        const type = definition.types && definition.types[configObject.type];
+        if (!type) {
+            throw new Error(`Unknown type: ${configObject.type}`);
+        }
+        const signature = type.signature;
+        if (!signature) {
+            throw new Error(`Type ${configObject.type} has no signature`);
+        }
+        const convertedType = gltfTypeToBabylonType[signature];
+        return {
+            value: configObject.value,
+            className: convertedType,
+        };
+    } else {
+        return configObject.value;
+    }
+}
+
+function convertConfiguration(gltfBlock: IKHRInteractivity_Node, definition: IKHRInteractivity): IFlowGraphBlockConfiguration {
+    const converted: IFlowGraphBlockConfiguration = {};
+    const configurationList: IKHRInteractivity_Configuration[] = gltfBlock.configuration ?? [];
+    for (const configObject of configurationList) {
+        if (configObject.id === "customEvent") {
+            const customEvent = definition.customEvents && definition.customEvents[configObject.value];
+            if (!customEvent) {
+                throw new Error(`Unknown custom event: ${configObject.value}`);
+            }
+            converted.eventId = customEvent.id;
+            converted.eventData = customEvent.values.map((v) => v.id);
+        } else if (configObject.id === "variable") {
+            const variable = definition.variables && definition.variables[configObject.value];
+            if (!variable) {
+                throw new Error(`Unknown variable: ${configObject.value}`);
+            }
+            converted.variableName = variable.id;
+        } else if (configObject.id === "path") {
+            // Convert from a GLTF path to a reference to the Babylon.js object
+            let pathValue = configObject.value as string;
+            if (!pathValue.startsWith("/")) {
+                pathValue = `/${pathValue}`;
+            }
+            // A path can be:
+            // /[nodes|materials|animations]/[numericIndex|substitutionString]/propertyName
+            // Basically the first two parts are part of the path, and from then on it's part of the property?

```

Comment:
As we discussed offline, we should factor out this parsing from the animation pointer extension and use the same function for this extension.